lib/most_frequent_integer.rb

```diff
@@ -13,4 +13,17 @@ def find_most_frequent_integer(arr)
 #If the value is unique, return its key
 #Else return nil
 
+new_hash = {}
+arr.each do |x|
+    new_hash[x] = 0 if new_hash[x].nil?
+    new_hash[x] = new_hash[x] + 1
+end
+new_hash2 = {}
+new_hash2 = new_hash.select{|k,v| v == new_hash.values.max}
+
+if new_hash2.length == 1
+    return new_hash2.key(2)
```

Comment:
`hash.key(value)` returns the key for the value passed in as a parameter. So, as it's written, `new_hash2.key(2)` assumes the value (or count of the numbers) will always be `2`.

I think that works alright for now because, in the tests file, there are no scenarios written for an array where the most-frequently occurring number is listed more than two times in the Array.

Since, in this scenario you know the length of the `new_hash2` Hash only contains one key-value pair, I would look into returning [a collection of the keys](http://ruby-doc.org/core-2.3.1/Hash.html#method-i-keys) in the Hash, knowing that the collection will also contain only one key.